### PR TITLE
MODKBEKBJ-385 - Implement endpoint to bulk load resources records

### PR DESCRIPTION
**Pull request by natali-zaitseva**
## Purpose
The issue addresses to https://issues.folio.org/browse/MODKBEKBJ-385 to add an endpoint to load multiple resources by one request to mod-kb-ebsco-java

## Approach
* modified resources.raml file  by adding endpoint for bulk load of resources /eholdings/resources/bulk/fetch
* updated ModuleDescriptor file by adding permission for resources bulk load endpoint
* added implementation for /eholdings/resources/bulk/fetch endpoint
* added tests to cover new functionality

#### TODOS and Open Questions
- [ ] discuss whether is it a good idea to limit the number of loaded at the same time items to 20?
 ` "maxItems": 20 `

### CODE REVIEW

File: src/main/java/org/folio/rest/converter/resources/ResourceCollectionBulkFetchConverter.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  private Converter<Title, ResourceBulkFetchCollectionItem> converter;
+
+  @Override
+  public ResourceBulkFetchCollection convert(ResourceBulkResult resourceBulkResult) {
+    List<ResourceBulkFetchCollectionItem> titleList = mapItems(resourceBulkResult.getTitles(), converter::convert);
+    return new ResourceBulkFetchCollection()
+      .withIncluded(titleList)
+      .withJsonapi(RestConstants.JSONAPI)
+      .withMeta(new MetaResourcesInformation()
+      .withFailed(new org.folio.rest.jaxrs.model.Failed()
```
Comment (psmagin) [inline src/main/java/org/folio/rest/converter/resources/ResourceCollectionBulkFetchConverter.java]:
```suggestion
      .withFailed(new Failed()
```

File: src/main/java/org/folio/rmapi/ResourcesServiceImpl.java [hunk 2]
```diff
@@ -111,4 +141,17 @@ private Titles mapToResources(List<ResourceResult> resourceFutures) {
       .titleList(titlesList)
       .build();
   }
+
+  public ResourceBulk parseToResourceId(Set<String> packagesSet, IdParser idParser) {
```
Comment (psmagin) [inline src/main/java/org/folio/rmapi/ResourcesServiceImpl.java]: Confusing naming.
```suggestion
  public ResourceBulk parseToResourceBulk(Set<String> resourceIdSet, IdParser idParser) {
```

File: src/main/java/org/folio/rmapi/result/ResourceBulkResult.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+import java.util.List;
+
+import lombok.AllArgsConstructor;
+import lombok.Value;
+
+import org.folio.holdingsiq.model.Title;
+
+@Value
+@AllArgsConstructor
```
Comment (psmagin) [inline src/main/java/org/folio/rmapi/result/ResourceBulkResult.java]: `@Value` already includes `@AllArgsConstructor`

Comment (natali-zaitseva) [reply src/main/java/org/folio/rmapi/result/ResourceBulkResult.java]: @psmagin thanks, fixed

**Pull request merged**